python/tk_maya/panel_generation.py

```diff
@@ -194,6 +209,16 @@ def build_workspace_control_ui(shotgun_panel_name):
     for widget in QtWidgets.QApplication.allWidgets():
         if widget.objectName() == shotgun_panel_name:
 
+            # When possible, give a minimum width to the workspace control;
+            # otherwise, it will use the width of the currently displayed tab.
+            # Note that we did not use the workspace control "initialWidth" and "minimumWidth"
+            # to set the minimum width to the initial width since these values are not
+            # properly saved by Maya 2017 in its layout preference files.
```

Comment:
might be worth adding here that this minimum width behaviour is consistent with maya's own std panels.